code/game/machinery/doors/brigdoors.dm

```diff
@@ -303,40 +355,72 @@
 // 	"change" resets the timer to the timetoset amount while the timer is counting down
 // Also updates dialog window and timer icon
 /obj/machinery/door_timer/Topic(href, href_list)
-	if(..())
-		return 1
+	..()
 
 	if(!allowed(usr) && !usr.can_admin_interact())
 		return 1
 
 	usr.set_machine(src)
 
-	if(href_list["timing"])
-		timing = text2num(href_list["timing"])
-
-		if(timing)
-			timer_start()
+	if(href_list["timingstart"])
+		if(!detaineecrimes | isnull(detaineename) | timetoset == 0)
+			visible_message("<span class='warning'>The [src] buzzes. Cannot activate without all information inputted.</span>")
+			playsound(loc, 'sound/machines/buzz-sigh.ogg', 50, 0)
 		else
-			timer_end()
-			Radio.autosay("Timer stopped manually by [usr.name].", name, "Security", list(z))
-
-	else
-		if(href_list["tp"])  //adjust timer, close door if not already closed
-			var/tp = text2num(href_list["tp"])
-			var/addtime = (timetoset / 10)
-			addtime += tp
-			addtime = min(max(round(addtime), 0), 3600)
-
+			timer_start()
+			alreadyrunning = 1
+			timing = 1
+
+	if(href_list["timingstop"])
+		switch(alert("Are you sure you wish to stop the cell timer? All data will be cleared", "Stop Cell Timer?", "Stop", "Abort"))
+			if("Stop")
+				Radio.autosay("Timer stopped manually by [usr.name].", name, "Security", list(z))
+				timer_end()
+				timing = 0
+			if("Abort")
+				return
+
+	if(href_list["tp"])  //adjust timer, close door if not already closed
+		var/tp = text2num(href_list["tp"])
+		var/addtime = (timetoset / 10)
+		addtime += tp
+		addtime = min(max(round(addtime), 0), 3600)
+		if(!timing)
 			timeset(addtime)
+		else
+			releasetime = (world.timeofday + timeset(addtime) * 10)
 
-		if(href_list["fc"])
-			for(var/obj/machinery/flasher/F in targets)
-				F.flash()
-
-		if(href_list["change"])
-			printed = 1
+	if(href_list["change"])
+		var/newtime = input(usr, "How many minutes would you like to set the timer to(minutes)?") as num

```

Comment:
Does this only allow you to set minutes? We are outputting sentences in seconds earlier, maybe this can be more consistent?